src/tests/no-incompatible-branch-types.test.coffee

```diff
@@ -0,0 +1,68 @@
+path = require 'path'
+
+{ESLintUtils} = require '@typescript-eslint/experimental-utils'
+
+{rules: {'no-incompatible-branch-types': rule}} = require '..'
+
+ruleTester = new ESLintUtils.RuleTester(
```

Comment:
Copied the pattern for testing a Typescript-specific ESLint rule from `eslint-plugin-reanimated`

I guess `ESLintUtils.RuleTester` is Jest-specific so I had to switch the test runner from Mocha -> Jest